inspirehep/modules/literaturesuggest/tasks.py

```diff
@@ -147,16 +148,16 @@ def formdata_to_model(obj, formdata):
         data['collections'].extend([{'primary': "ConferencePaper"}])
 
     # ============================
-    # Page range
+    # Page number
     # ============================
     if 'page_nr' not in data:
-        if data.get("publication_info", [{}])[0].get("page_artid"):
-            pages = data['publication_info'][0]['page_artid'].split('-')
-            if len(pages) == 2:
-                try:
-                    data['page_nr'] = int(pages[1]) - int(pages[0]) + 1
-                except ValueError:
-                    pass
+        if (data.get("publication_info", [{}])[0].get("page_start") and
+                data.get("publication_info", [{}])[0].get("page_end")):
```

Comment:
Explainer variables would improve this:
```python
first_publication_info = data.get('publication_info', [{}])[0]
page_start = first_publication_info.get('page_start')
page_end = first_publication_info.get('page_end')

if page_start and page_end:
    try:
        data['page_nr'] = page_end - page_start + 1
    except ValueError:
        pass
```
which suggests that this code never throws a `ValueError` after all.